valhalla/app/javascript/mixins/manifesto-filemanager-mixins.js

```diff
@@ -0,0 +1,80 @@
+const ManifestoFilemanagerMixins = {
+
+  mainSequence: function () {
+    // mainSequence is the one without an id (not ideal since it could have an id)
+    const s = this.getSequences()
+    const main_sequence = s.filter((seq) => seq.id !== 'undefined')
+    return main_sequence[0]
+  },
+
+  getCanvasMainThumb: function (canvas) {
+    const images = canvas.getImages()
+    var thumb = "http://via.placeholder.com/300x400"
+    var services = images[0].getResource().getServices()
+    if (services.length) {
+      thumb = services[0].id + '/full/400,/0/default.jpg'
+    }
+    return thumb
+  },
+
+  getResourceId: function (canvas) {
+    const images = canvas.getImages()
+    const r = images[0].getResource()
+    // this seems to be a bug in manifesto ...
```

Comment:
Can you open a ticket about this? It would be good to get some clarification on whether this is the expected behavior or not.

Reply:
It's actually a bug in our code. Tom Crane confirmed that it is an invalid manifest. I opened ticket #598. I will update this code comment to reflect that.